nexus/types/src/deployment/planning_input.rs

```diff
@@ -240,6 +254,24 @@ impl PlanningInput {
         self.policy.oximeter_read_policy.mode.single_node_enabled()
     }
 
+    /// ID of the currently running Nexus zones
+    pub fn active_nexus_zones(&self) -> &BTreeSet<OmicronZoneUuid> {
+        &self.active_nexus_zones
+    }
+
+    pub fn set_active_nexus_zones(&mut self, ids: BTreeSet<OmicronZoneUuid>) {
+        self.active_nexus_zones = ids;
+    }
```

Comment:
Thus far, there are no setters in `PlanningInput`.  Instead, callers use `into_builder()` to get a mutable thing, set what they want, and go back to `PlanningInput`.  I prefer that, since it feels like an input struct should be immutable.  (I guess one could argue it's redundant since rust has `mut` to tell you when something is mutable.)

Reply:
Done in 1af0b04b.

This impacts #8936 more, and the equivalent changes are made in https://github.com/oxidecomputer/omicron/pull/8936/commits/02d6aae9944ee0946079650c78fe2b7f2fb414e1